demisto_sdk/commands/test_content/xsoar_tools/xsoar_client.py

```diff
@@ -0,0 +1,647 @@
+import os
+import re
+import urllib.parse
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional, Union
+
+import dateparser
+import demisto_client
+import requests
+from demisto_client.demisto_api.models import (
+    IncidentWrapper,
+)
+from demisto_client.demisto_api.rest import ApiException
+from pydantic import BaseModel, Field, SecretStr, validator
+from pydantic.fields import ModelField
+from requests.auth import HTTPBasicAuth
+
+from demisto_sdk.commands.common.logger import logger
+from demisto_sdk.utils.utils import retry_http_request
+
+
+class XsoarApiClientConfig(BaseModel):
+    base_url: str = Field(
+        default=os.getenv("DEMISTO_BASE_URL"), description="XSOAR Tenant Base API URL"
+    )
+    api_key: SecretStr = Field(
+        default=SecretStr(os.getenv("DEMISTO_API_KEY", "")), description="XSOAR API Key"
+    )
+
+    @validator("base_url", "api_key", always=True)
+    def validate_client_config(cls, v, field: ModelField):
+        if not v:
+            raise ValueError(
+                f"XSOAR client configuration is not complete: value was not passed for {field.name} and"
+                f" the associated environment variable for {field.name} is not set"
+            )
+        return v
+
+
+class XsoarNGApiClientConfig(XsoarApiClientConfig):
+    auth_id: Optional[str] = Field(
+        default=os.getenv("XSIAM_AUTH_ID"), description="XSOAR-NG Auth ID"
+    )
+
+    @validator("auth_id", always=True)
+    def validate_client_config(cls, v, field: ModelField):
```

Comment:
no need, pynadic automatically trigger the `validate_client_config` of all class hierarchy
no special reason to override, its just mutual fields that can be used by xsoar client (not xsoar-ng) when it will be implemented in the future

Reply:
pydantic triggers any function with the validator decorator automatically.